### PR TITLE
Typos

### PR DESCRIPTION
**Pull request by deadcoder0904**

### CODE REVIEW

File: recipes/lifecycle-hooks.md [hunk 1]
```diff
@@ -49,9 +49,9 @@ class _MyComponentState extends State<MyComponent> {
   @override
   void didUpdateWidget(MyComponent oldWidget) {
     // this method IS called when parent widget passed new "props"
-    // unlike React this method IS callled _before_ the build
-    // unlike React method receives OLD "props"
-    // unlike React this method ISN'T called after setState()
+    // unlike React, this method IS called _before_ the build
+    // unlike React, this method receives OLD "props"
```
Comment (deadcoder0904) [inline recipes/lifecycle-hooks.md]: Bdw I didn't understand this line

Comment (lesnitsky) [reply recipes/lifecycle-hooks.md]: You are right, initially it was `componentWillUpdate` on react side, but it was deprecated and I decided to use `componentDidUpdate`, so this explanation comment is redundant, but it worth replacing it with one mentioning that `didUpdateWidget` is called before `build`

File: recipes/arrays.md [hunk 2]
```diff
@@ -27,7 +27,7 @@ items.forEach((item, index) => console.log(item, index));
 
 const includes = items.includes(42);
 
-const indexOf = items.indexOf(item => item === itemToFind);
+const indexOf = items.indexOf(item => item === 42);
```
Comment (lesnitsky) [inline recipes/arrays.md]: `42` is always better 👍

File: recipes/arrays.md [hunk 3]
```diff
@@ -42,7 +42,7 @@ const reduced = items.reduce((acc, item) => {
   return acc;
 }, {});
 
-items.reverse();
+const reversed = items.reverse();
```
Comment (lesnitsky) [inline recipes/arrays.md]: the intent here was to accentuate that `.reverse()` doesn't return new collection, but rather modifies original in place. On the other had dart example has an assignment, so perhaps it worth keeping this example consistent on both sides

**Comment by lesnitsky** [comment]
@deadcoder0904 thanks 👍

**Pull request merged**